terraform/e2e/gclb.tf

```diff
@@ -0,0 +1,219 @@
+# Copyright 2023 The Authors (see AUTHORS file)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+resource "google_compute_global_address" "default" {
+  project    = var.project_id
+  name       = "jvs-${random_id.default.hex}-address" # 63 character limit
+  ip_version = "IPV4"
+
+  depends_on = [
+    google_project_service.services["compute.googleapis.com"],
+  ]
+}
+
+resource "google_compute_global_forwarding_rule" "http" {
+  project               = var.project_id
+  name                  = "jvs-${random_id.default.hex}-http" # 63 character limit
+  target                = google_compute_target_http_proxy.default.self_link
+  ip_address            = google_compute_global_address.default.address
+  port_range            = "80"
+  load_balancing_scheme = "EXTERNAL"
+}
+
+resource "google_compute_global_forwarding_rule" "https" {
+  project               = var.project_id
+  name                  = "jvs-${random_id.default.hex}-https" # 63 character limit
+  target                = google_compute_target_https_proxy.default.self_link
+  ip_address            = google_compute_global_address.default.address
+  port_range            = "443"
+  load_balancing_scheme = "EXTERNAL"
+}
+
+resource "google_compute_managed_ssl_certificate" "default" {
+  project = var.project_id
+  name    = "jvs-${random_id.default.hex}-cert" # 63 character limit
+
+  managed {
+    domains = toset([var.jvs_api_domain, var.jvs_ui_domain])
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  depends_on = [
+    google_project_service.services["compute.googleapis.com"],
+  ]
+}
+
+resource "google_compute_url_map" "default" {
+  project         = var.project_id
+  name            = "jvs-${random_id.default.hex}-url-map" # 63 character limit
+  default_service = google_compute_backend_service.jvs_api_backend.self_link
+
+  host_rule {
+    hosts        = [var.jvs_api_domain]
+    path_matcher = "jvs-api"
+  }
+
+  host_rule {
+    hosts        = [var.jvs_ui_domain]
+    path_matcher = "jvs-ui"
```

Comment:
Is this redundant if url is `https://jvs.domain.com/jvs-ui`?

Should it just be `https://jvs.domain.com/ui`?

Reply:
@shan4993 - In a brief discussion, we thought it's clearer to have `https://ui.jvs.domain.com/popup` than `https://jvs.domain.com/ui/popup`. In this dev env, I'm fine with either way.

Reply:
I mis-read this. This is fine.